nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/certificate/CertificateRealm.java

```diff
@@ -175,6 +210,39 @@ public String authenticate(Subject subject, X500Principal principal) {
         return principalName;
     }
 
+    private boolean validateSubjectViaAPI(Subject subject, X500Principal principal, X509Certificate certificate) {
+        try {
+            return loadVetoClasses().stream().anyMatch(veto -> !veto.isValid(subject, principal, certificate));
+        } catch (Throwable exc) {
+            _logger.log(Level.WARNING, "Exception from certificate validation class", exc);
+            clientCertificateValidatorMap.remove(Utility.getClassLoader());
+        }
+        return false;
+    }
+
+    private List<ClientCertificateValidator> loadVetoClasses() {
```

Comment:
And shouldn't this return the list of interface?

Reply:
Again, you are confused. it is returning a list of interfaces.

Reply:
Yes, you are right.